File: 6-kyu/build-a-pile-of-cubes/c++/solution.cpp
#include <cmath>

class ASum
{
  public:
  static long long findNb(long long m);
};

long long ASum::findNb(long long m)
{
    long long ret = 0LL;
    while (true) {
        ret++;
        m -= static_cast<long long>(std::pow(ret, 3));
        if (m == 0) return ret;
        else if (m < 0) return -1;
    }
}

File: 6-kyu/build-a-pile-of-cubes/swift/solution.swift
import Foundation

func findNb(_ number: Int) -> Int {
    var ret = 0, nb = number
    while true {
        ret += 1
        nb -= Int(pow(Double(ret), 3.0))
        if nb == 0 { return ret }
        else if nb < 0 { return -1 }
    }
}